File: AndroidQuiz51/app/src/main/java/com/example/androidquiz51/MainActivity.java
package com.example.androidquiz51;

import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.EditText;

import org.w3c.dom.Text;

public class MainActivity extends AppCompatActivity {
    EditText monday;
    EditText tuesday;
    EditText wednesday;
    EditText thursday;
    EditText friday;
    SharedPreferences P;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        P = getPreferences(MODE_PRIVATE);
        setContentView(R.layout.activity_main);
        monday = (EditText)findViewById(R.id.monday);
        tuesday = (EditText)findViewById(R.id.tuesday);
        wednesday = (EditText)findViewById(R.id.wednesday);
        thursday = (EditText)findViewById(R.id.thursday);
        friday = (EditText)findViewById(R.id.friday);
    }

    protected void onStart()
    {
        super.onStart();
        monday.setText(P.getString("monday",""));
        tuesday.setText(P.getString("tuesday",""));
        wednesday.setText(P.getString("wednesday",""));
        thursday.setText(P.getString("thursday",""));
        friday.setText(P.getString("friday",""));
    }
    protected void onStop()
    {
        SharedPreferences.Editor editor = P.edit();
        editor.putString("monday",monday.getText().toString());
        editor.putString("tuesday",tuesday.getText().toString());
        editor.putString("wednesday",wednesday.getText().toString());
        editor.putString("thursday",thursday.getText().toString());
        editor.putString("friday",friday.getText().toString());

        editor.commit();
        super.onStop();
    }
}

File: MyApplication2/app/src/main/java/com/example/MainActivity.kt
package com.example
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.myapplication2.R
import android.view.View
import android.widget.EditText
import android.os.Parcel
import android.os.Parcelable

class MainActivity() : AppCompatActivity(), Parcelable {
    constructor(parcel: Parcel) : this() {
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val euros = findViewById<View>(R.id.euros) as EditText
        val bitcoin = findViewById<View>(R.id.bitcoin) as EditText
        euros.setOnEditorActionListener { textView, i, keyEvent ->
            val euros = textView.text.toString().toDouble()
            bitcoin.setText(java.lang.Double.toString(euros * 2))
            true
        }
        bitcoin.setOnEditorActionListener { textView, i, keyEvent ->
            val bitcoin = textView.text.toString().toDouble()
            euros.setText(java.lang.Double.toString(bitcoin / 2))
            true
        }
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {

    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<MainActivity> {
        override fun createFromParcel(parcel: Parcel): MainActivity {
            return MainActivity(parcel)
        }

        override fun newArray(size: Int): Array<MainActivity?> {
            return arrayOfNulls(size)
        }
    }
}